plugins/imjournal/imjournal.c

```diff
@@ -912,6 +920,8 @@ CODESTARTsetModCnf
 			cs.bUseJnlPID = (int) pvals[i].val.d.n;
 		} else if (!strcmp(modpblk.descr[i].name, "usepid")) {
 			cs.usePid = (char *)es_str2cstr(pvals[i].val.d.estr, NULL);
+		} else if (!strcmp(modpblk.descr[i].name, "workaroundjournalbug")) {
+			cs.bWorkAroundJournalBug = (int) pvals[i].val.d.n;
```

Comment:
What is the default value for `cs.bWorkAroundJournalBug`?  Where is it set?

Reply:
My understanding is that default value for binary arguments is `off`, unless set otherwise (see for example other binary arguments).

Reply:
actually depends on the initialization. The journal plugins use the old-style cs. structure, which is something we need to change at some point in time but so far I didn't care much about it. I'll also do a final review on the code, but check for where the other settings are initialized (well, they should, maybe it's missing altogether).

Reply:
Well, when I look on the other binary arguments I do not see explicit initialization in any of them (only set on them is from the params parsing), but as implementation wise they are 'static int' they are guaranteed to be zero by standard (unless I understood current code wrong).

Reply:
yeah... that's actually the case. But I am not sure if it is guaranteed by the standard. At least in the past there were some platforms that did not zero out that memory. They may long be history...

From an "esay to read" PoV it may be better though to initialize them. If only it saves discussions ;-) But I understand that is something you inherited. There are quite some things in that module that are a bit diffrent than in the rest of rsyslog...

Reply:
I have no problem adding the explicit initialization for the sake or readability (will look to other module for example). However `static` zero-initialization *is* part of the C standard (see for example [here](http://www.open-std.org/jtc1/sc22/WG14/www/docs/n1256.pdf#%5B%7B%22num%22%3A272%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C-27%2C816%2Cnull%5D) - point 10. Of course that may be minus some non-standard compliant compilers :-)

Reply:
@jvymazal thx, I hadn't the reference at hand.